modules/web/services/darkhttpd/default.nix

```diff
@@ -0,0 +1,206 @@
+	therefore turned off by default.
+      '';
+    };
+    log = mkOption {
+    default = false;
+      description = ''
+        Whether to log any data about requests to the webserver,
+	or not. If you are not going to use it, because you use
+	some other mechanism (like Piwik) you might as well keep it
+	turned off to save yourself disk wear and enhance user privacy.
+      '';
+    };
+    logFile = mkOption {
+    default = "/var/log/${serviceName}";
+      description = ''
+	Location of the file where the logs are stored. If you want
+	to log anything that is.
+      '';
+    };
+    serviceName = mkOption {
+    default = serviceName;
+    };
+};
+
+# Serving only on localhost for security purposes. The ports do not have to be
+# reachable from the outside directly, as the static site will be proxied anyway.
+# If you would want to serve on the actual IP address, use
+# "${config.proxyOptions.ip}" instead of "127.0.0.1";
+
+  meta = {
+    description = "An efficient, very restricted service for serving up HTML";
+    longDescription = ''
+This is a simple and secure service for a static website using the
+nixcloud webservice abstraction. A basic configuration.nix would
+for example contain the following:
+
+  nixcloud.webservices.darkhttpd = {
+    foo.enable = true;
+    foo.staticDir = "/path/to/statictest/";
+    foo.proxyOptions.port = 8087;
+    foo.proxyOptions.domain = "example.net";
+  };
+
+This would serve the HTML content in /path/to/statictest as a website on
+http(s)://example.net, proxied from a local server listening on port 8087. If
+you do not include the value for staticDir, you will be assigned a standard
+location.
+
+NOTE THAT /TMP IS NOT CONSIDERED A VALID LOCATION, AS A SEPARATE /TMP TEMPORARY
+directory will be mounted (read-only) to isolate the server process from
+anything else happening on the computer. While we are dealing with static HTML,
+we don't know if the server needs temporary files in any way and by using a
+separate temporary directory we protect from any potential data leaks resulting
+from that.
+
+There are some fancier options one can use, such as:
+
+    foo.serviceOwner = "roman";
+    foo.proxyOptions.TLS = "none";
+    foo.proxyOptions.http.mode = "on";
+    foo.proxyOptions.https.mode = "off";
+    foo.proxyOptions.path = "/"; };
+
+'';
+
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ leenaars ];
+    meta.platforms = lib.platforms.linux;
+  };
+
+   config = let
+    serviceCommand = "${pkgs.${config.serviceName}}/bin/${serviceName}";
+    serviceParameters = [
+      "${toString config.staticDir}"
+      "--addr" "127.0.0.1"
+      "--port" "${toString config.proxyOptions.port}"
+      "--no-server-id"
+      "--daemon"
+      "--no-keepalive"
+      "--mimetypes" "${pkgs.apacheHttpd}/conf/mime.types"
+    ] ++ optional (!config.directoryListing) "--no-listing"
+      ++ optional (config.log) "--log ${config.logFile}";
+  in mkIf config.enable rec {
```

Comment:
Making this whole block recursive is a good way to shoot yourself in the foot, especially because I can't see any attribute that is benefiting from it here.